.github/workflows/binary.yaml

```diff
@@ -104,6 +102,35 @@ jobs:
             dune pkg lock
             dune build
 
+  build_with_dune:
+    needs: binary
```

Comment:
**fix**. can we install the last version of the binary to build the new version of the binary? that'd simplify this and allow them to run concurrently too.